src/coreclr/vm/peimagelayout.cpp

```diff
@@ -251,18 +259,21 @@ void PEImageLayout::ApplyBaseRelocations(bool relocationMustWriteCopy)
 #if defined(TARGET_UNIX)
                 if (((pSection->Characteristics & VAL32(IMAGE_SCN_MEM_EXECUTE)) != 0))
                 {
-#ifdef __APPLE__
-                    dwNewProtection = PAGE_READWRITE;
```

Comment:
@jkotas Note that this was introduced in #61938 for fixing the loading of R2R images on Apple Silicon. I believe this was never needed for Intel, hence why I removed it here. @VSadov could you please confirm that this sounds right to you?